src/uri/path.rs

```diff
@@ -169,6 +165,27 @@ impl PathAndQuery {
 
         ret
     }
+    /// Returs the segments of path component
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use http::uri::*;
+    ///
+    /// let path_and_query: PathAndQuery = "/hello/world".parse().unwrap();
+    ///
+    /// assert!(path_and_query.path_segments().is_some());
+    /// assert_eq!(path_and_query.path_segments().unwrap().collect::<Vec<_>>(), vec!["hello",
+    /// "world"])
+    /// ```
+    pub fn path_segments(&self) -> Option<str::Split<char>> {
```

Comment:
This should probably return a custom iterator: `struct PathSegments<str::Split<...>>`

Is `Option` significant in this case or can the iterator be returned directly? `Uri::path_and_query` already returns an `Option`.

Reply:
You're right, `Option` in this context is meanless.

Reply:
And custom iterator you said, I need help.. I'm so new to rust, When I write any custom iterator, compiler raise lifetime error. T^T